pkg/tar/tar.go

```diff
@@ -338,6 +338,7 @@ func (t *stiTar) ExtractTarStreamFromTarReader(dir string, tarReader Reader, log
 					glog.Errorf("Error creating dir %q: %v", dirPath, err)
 					return err
 				}
+				t.Chmod(dirPath, header.FileInfo().Mode())
```

Comment:
note that i did this instead of changing the 0700 above because i only wanted to affect the permissions of files/dirs explicitly defined in the tar file, not parent dirs being indirectly created.